manifests/init.pp

```diff
@@ -12,12 +12,20 @@
   $localzonepath        = $::dns::params::localzonepath,
   $forwarders           = $::dns::params::forwarders,
   $listen_on_v6         = $::dns::params::listen_on_v6,
+  $recursion            = $::dns::params::recursion,
   $allow_recursion      = $::dns::params::allow_recursion,
+  $allow_query          = $::dns::params::allow_query,
+  $dnssec_enable        = $::dns::params::dnssec_enable,
+  $dnssec_validation    = $::dns::params::dnssec_validation,
   $namedconf_template   = $::dns::params::namedconf_template,
   $optionsconf_template = $::dns::params::optionsconf_template,
 ) inherits dns::params {
   validate_array($dns::forwarders)
   validate_array($dns::allow_recursion)
+  validate_array($dns::allow_query)
+  validate_re($dns::recursion, ['yes', 'no'], 'Only \'yes\' and \'no\' are valide values for recursion field')
```

Comment:
a) use ^ and $ in regexes to anchor it, e.g. `'^yes$'` or even combine them with '^(yes|no)$'.
b) "valide" should be "valid"